themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/applications-page/ApplicationsPage.tsx

```diff
@@ -15,21 +15,196 @@
- 
+import { AxiosResponse } from 'axios';
+
+import {
+  DataList,
+  DataListItem,
+  DataListItemRow,
+  DataListCell,
+  DataListToggle,
+  DataListContent,
+  DataListItemCells,
+  Text,
+  TextVariants,
+  Grid,
+  GridItem,
+} from '@patternfly/react-core';
+
+import { InfoAltIcon, CheckIcon, LinkIcon, BuilderImageIcon } from '@patternfly/react-icons';
+import { ContentPage } from '../ContentPage';
+import { ContinueCancelModal } from '../../widgets/ContinueCancelModal';
+import { AccountServiceClient } from '../../account-service/account.service';
+
 export interface ApplicationsPageProps {
 }
- 
-export class ApplicationsPage extends React.Component<ApplicationsPageProps> {
-    
-    public constructor(props: ApplicationsPageProps) {
-        super(props);
-    }
-
-    public render(): React.ReactNode {
-        return (
-            <div>
-              <h2>Hello Applications Page</h2>
-            </div>
-        );
-    }
-};
\ No newline at end of file
+
+export interface ApplicationsPageState {
+  isRowOpen: boolean[];
+  applications: Application[];
+}
+
+export interface GrantedScope {
+  displayTest: string;
+  id: string;
+  name: string;
+}
+
+export interface Consent {
+  createDate: number;
+  grantedScopes: GrantedScope[];
+  lastUpdatedDate: number;
+}
+
+interface Application {
+  baseUrl: string;
+  clientId: string;
+  clientName: string;
+  consent: Consent;
+  description: string;
+  inUse: boolean;
+  offlineAccess: boolean;
+  userConsentRequired: boolean;
+  scope: string[];
+}
+
+export class ApplicationsPage extends React.Component<ApplicationsPageProps, ApplicationsPageState> {
+
+  public constructor(props: ApplicationsPageProps) {
+    super(props);
+    this.state = {
+      isRowOpen: [],
+      applications: []
+    };
+
+    this.fetchApplications();
+  }
+
+  private removeConsent = (clientId: string) => {
+    AccountServiceClient.Instance.doDelete("/applications/" + clientId + "/consent")
+      .then(() => {
+        this.fetchApplications();
+      });
+  }
+
+  private onToggle = (row: number): void => {
+    const newIsRowOpen: boolean[] = this.state.isRowOpen;
+    newIsRowOpen[row] = !newIsRowOpen[row];
+    this.setState({ isRowOpen: newIsRowOpen });
+  };
+
+  private fetchApplications(): void {
+    AccountServiceClient.Instance.doGet("/applications")
+      .then((response: AxiosResponse<Object>) => {
+        const applications = response.data as Application[];
+        this.setState({
+          isRowOpen: this.collapseRows(applications),
+          applications: applications
+        });
+      });
+  }
+
+  private collapseRows(applications: Application[]): boolean[] {
+    const openRows: boolean[] = new Array<boolean>().fill(true);
+    applications.forEach((application: Application, appIndex: number) => {
+      openRows[appIndex] = false;
+    });
+    return openRows;
+  }
+
+  public render(): React.ReactNode {
+    return (
+      <ContentPage title="applications">
```

Comment:
General comment: we're missing internationalization support (all strings are hardcoded).

Reply:
+1

Reply:
@abstractj If this is out of scope of this PR, please create a sub-task here: https://issues.jboss.org/browse/KEYCLOAK-12338

Reply:
@vmuzikar I don't think it's out of scope and you're right on this. I can revisit it.